codegen/masm/src/codegen.rs

```diff
@@ -58,15 +64,18 @@ pub struct CodeGenerator<'ast> {
     config: CodegenConfig,
 }
 
-/// Converts a column ID to an element position.
+/// Given a periodic column group position, returns a memory offset.
 ///
-/// The element position is then used to determine the memory location to load, and which elements
-/// of the word must be kept. Values of periodic columns are stored at distinct memory addresses
-/// such that each value occupies a single memory word with the two most significant word elements
-/// set to zeros (i.e., [q0, q1, 0, 0])
-fn periodic_column_to_target_el(column: u32) -> u32 {
-    // each period column has its own address, and the element is in the lower half
-    (column * 2) + 1
+/// Periodic columns are grouped based on their length, this is done so that only a single z value
+/// needs to be cached per group. The grouping is based on unique lengths, sorted from highest to
+/// lowest. Given a periodic group, this function will return a memory offset, which can be used to
+/// load the correspoding z value.
+fn periodic_group_to_memory_offset(group: u32) -> u32 {
+    // Each memory address contains a single quadraic element, this makes the code to store/load the
```

Comment:
minor nit: element of a quadratic field extension

Reply:
fixed :)